sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ImplementIDisposableCorrectly.cs

```diff
@@ -48,203 +48,239 @@ public class ImplementIDisposableCorrectly : SonarDiagnosticAnalyzer
 
         protected sealed override void Initialize(SonarAnalysisContext context)
         {
-            context.RegisterSyntaxNodeActionInNonGenerated(CheckIDisposableImplementation, SyntaxKind.ClassDeclaration);
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var checker = new DisposableChecker((ClassDeclarationSyntax)c.Node, c.SemanticModel);
+
+                if (!checker.IsCompliant())

```

Comment:
Having method named IsXXX() is pretty common in the framework. Not sure what other name would be good.